client/src/components/MyClubs.tsx

```diff
@@ -1,10 +1,117 @@
+import React, { useState, useEffect } from 'react';
 import NavBar from "./NavigationBar";
 
 export default function MyClubs() {
+  const initialClubs = [
+    { name: 'Tennis Club', id: 'tennis' },
+    { name: 'Chess Club', id: 'chess' },
+    { name: 'Reading Club', id: 'reading' },
+    { name: 'Science Club', id: 'science' }
+  ];
+
+  const [clubs, setClubs] = useState(initialClubs);
+  const [leftClubs, setLeftClubs] = useState([]);
+  const [query, setQuery] = useState('');
+
+  // Function to handle leaving a club
+  const leaveClub = (clubId) => {
+    const leftClub = clubs.find(club => club.id === clubId);
+    setClubs(clubs.filter(club => club.id !== clubId));
+    setLeftClubs([...leftClubs, leftClub]);
+
+    // Set a timer to automatically remove the undo option after 30 seconds
+    setTimeout(() => {
+      setLeftClubs(leftClubs => leftClubs.filter(club => club.id !== clubId));
+    }, 5000); // 30 seconds
+  };
+
+  // Function to handle undoing leave action
+  const undoLeave = (clubId) => {
```

Comment:
Do we really want an undo option for this release? We can discuss it as a group but I think it's something to work on if we have extra time at the end of the year, there's more key functionalities to focus on currently IMO, just leaving a club is fine for now

Reply:
I don't see any need for this feature. You can simply just rejoin from the AllClubs page

Reply:
I thought that if the user accidentally clicked the leave button for the wrong club than it would be easier to have an undo feature instead of trying to search for the club in the AllClubs pages.

Reply:
I'd agree with Tyler, don't think we need this tbh

Reply:
I removed it